scripts/dts/dts_cmake.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# This script uses edtlib and a pickle edt to generate a CMake file which
+# describes the devicetree as CMake propoerties that can then be used in the
+# rest of the build system.
+
+import argparse
+import os
+import pickle
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(__file__), 'python-devicetree',
+                             'src'))
+
+
+def parse_args():
+    # Returns parsed command-line arguments
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--cmake-out", required=True,
+                        help="path to write the CMake property file")
+    parser.add_argument("--edt-pickle", required=True,
+                        help="path to read the pickled edtlib.EDT object from")
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    edt = pickle.load(open(args.edt_pickle, 'rb'))
```

Comment:
This is a weird nit, but I try to avoid assuming CPython like this when I can. Lines like this work in CPython because it's got a refcount-based garbage collector, so as soon as the reference to the open file gets dropped on the next line, it gets closed. But other Pythons don't have that behavior, so it's cleaner to use a context manager.

```py
with open(args.edt_pickle), 'rb') as f:
    edt = pickle.load(f)
```

This is the pattern we use everywhere else in zephyr.

Reply:
Fixed